module/VuFind/src/VuFind/Related/Bookplate.php

```diff
@@ -0,0 +1,242 @@
+<?php
+/**
+ * Related Records: Bookplates
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2009.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Related_Records
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:related_records_modules Wiki
+ */
+namespace VuFind\Related;
+
+/**
+ * Related Records: Bookplates
+ *
+ * @category VuFind
+ * @package  Related_Records
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:related_records_modules Wiki
+ */
+class Bookplate implements RelatedInterface
+{
+    /**
+     * Bookplate config
+     */
+    protected $config;
+
+    /**
+     * Data fields (usually Solr)
+     */
+    protected $fields;
+
+    /**
+     * Bookplate strings
+     */
+    protected $bookplateStrs;
+
+    /**
+     * Bookplate image names or full URLs
+     */
+    protected $bookplateImages;
+
+    /**
+     * Bookplate thumbnail image names or thumbnail URLs
+     */
+    protected $bookplateThumbnails;
+
+    /**
+     * URL template for full bookplate
+     */
+    protected $fullUrlTemplate;
+
+    /**
+     * URL temlate for thumbnail
+     */
+    protected $thumbUrlTemplate;
+
+    /**
+     * Display bookplate titles?
+     */
+    protected $displayTitles;
+
+    /**
+     * Constructor
+     *
+     * @param VuFind\Config\PluginManager $configLoader PluginManager
+     */
+    public function __construct(\VuFind\Config\PluginManager $configLoader)
+    {
+        $this->configLoader = $configLoader;
+        $this->config = $this->configLoader->get('Related');
```

Comment:
I think the part of my suggestions that may have gotten lost in transition was the idea that the settings passed to the related module via config.ini could be used to control where additional settings are loaded from. It looks like you've created a new Related.ini file which is hard-coded here. I would suggest that instead of loading the configuration in the constructor here, you instead do it in the `init()` method, so that you can load a different config file/section depending on the user's settings.

If it's still not clear what I'm looking for, I can take a stab at implementing it here so you can see -- just let me know if that would be helpful!

Also, it looks like the comment documenting the existence of this new module still needs to be added to config.ini around here: https://github.com/vufind-org/vufind/blob/dev/config/vufind/config.ini#L1764 -- not sure if that was here previously and got lost in refactoring or if I forgot to suggest it on the first pass. Either way, we should make sure it's there before we merge.

Reply:
I see what you're asking for. I took a stab at implementing it. This is now more flexible. The only hitch I ran into was that I couldn't get anything after the second `:` to pass through. Switching it to a `;` worked fine though.

Yes, I had the existence of the module documented in the beginning but lost this in refactoring. I added this back and moved the default bookplate config back to the same section.

Reply:
Interesting; since none of the existing Related modules use parameters (meaning this has never really been tested), I wonder if there's a bug in the code that's exploding on more `:` characters than necessary. When I have a bit more time, I'll investigate and fix it if necessary!

Reply:
@bbusenius, as I suspected, I think the problem lies here:

https://github.com/vufind-org/vufind/blob/dev/module/VuFind/src/VuFind/View/Helper/Root/Related.php#L114

That line should be changed to `$parts = explode(':', $current, 2);` and then the ':' separator should work for you.

Would you be willing to make that change here, switch the explode character back from ';' to ':' (for consistency with recommendation modules), and see if that sorts everything out? If it does, then we should adjust the comments in the .ini file to include the optional parameters.

If you don't have time, let me know and I can try to work on this when I carve out a little more of my own... :-)

Reply:
Thanks @demiankatz! I appreciate your help with diagnosing this. I made these changes. Let me know if there's anything else.